### PR TITLE
Upgrade vmware extension to 2023-09-01 API version 

### PR DESCRIPTION
**Pull request by jonathanhe-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [ ] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by jonathanhe-msft** [comment]
@kairu-ms can you review this PR? would like to have it included in the next CLI release

File: src/vmware/azext_vmware/_help.py [hunk 1]
```diff
@@ -298,6 +298,19 @@
       text: az vmware datastore disk-pool-volume create --name iSCSIDatastore1 --resource-group MyResourceGroup --cluster Cluster-1 --private-cloud MyPrivateCloud --target-id /subscriptions/11111111-1111-1111-1111-111111111111/resourceGroups/ResourceGroup1/providers/Microsoft.StoragePool/diskPools/mpio-diskpool/iscsiTargets/mpio-iscsi-target --lun-name lun0
 """
 
+helps['vmware datastore elastic-san-volume'] = """
+    type: group
+    short-summary: Create an Elastic SAN volume in a private cloud cluster using Microsoft.ElasticSan provider.
```
Comment (yanzhudd) [inline src/vmware/azext_vmware/_help.py:303]: It is a command group, please check this short summary.
```suggestion
    short-summary: Manage Elastic SAN volume resource.

```

Comment (jonathanhe-msft) [reply src/vmware/azext_vmware/_help.py]: Fixed and updated similar issues.

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**